x-pack/plugins/threat_intelligence/public/containers/integrations_guard/integrations_guard.tsx

```diff
@@ -7,26 +7,75 @@
 
 import { EuiLoadingLogo } from '@elastic/eui';
 import React, { FC } from 'react';
+import { HttpSetup } from '@kbn/core/public';
+import { useRequest, UseRequestConfig } from '@kbn/es-ui-shared-plugin/public';
 import { EmptyPage } from '../../modules/empty_page';
 import { useIndicatorsTotalCount } from '../../modules/indicators';
 import { SecuritySolutionPluginTemplateWrapper } from '../security_solution_plugin_template_wrapper';
 
+type IntegrationInstallStatus = 'installed' | 'installing' | 'install_failed';
+
+let httpClient: HttpSetup;
+
+const PACKAGES_CONFIG: UseRequestConfig = {
+  path: '/api/fleet/epm/packages',
+  method: 'get',
+};
+
+export const THREAT_INTELLIGENCE_CATEGORY = 'threat_intel';
+
+export const THREAT_INTELLIGENCE_UTILITIES = 'ti_util';
+
+export const installationStatuses = {
+  Installed: 'installed',
+  Installing: 'installing',
+  InstallFailed: 'install_failed',
+  NotInstalled: 'not_installed',
+};
+
+interface Integration {
+  categories: string[];
+  id: string;
+  status: IntegrationInstallStatus;
+}
+
+export const setHttpClient = (client: HttpSetup) => {
+  httpClient = client;
+};
+
 /**
- * Renders children only if TI integrations are enabled
+ * Renders children only if TI integrations are enabled and indicators are received
  */
 export const IntegrationsGuard: FC = ({ children }) => {
-  const { count: indicatorsTotalCount, isLoading: isIndicatorsTotalCountLoading } =
+  const { isLoading: indicatorsTotalCountLoading, count: indicatorsTotalCount } =
     useIndicatorsTotalCount();
 
-  if (isIndicatorsTotalCountLoading) {
+  const { isLoading: packagesLoading, data: integrations } = useRequest(
+    httpClient,
+    PACKAGES_CONFIG
+  );
+
+  if (packagesLoading && indicatorsTotalCountLoading) {
     return (
       <SecuritySolutionPluginTemplateWrapper isEmptyState>
         <EuiLoadingLogo logo="logoSecurity" size="xl" />
       </SecuritySolutionPluginTemplateWrapper>
     );
   }
 
-  const showEmptyPage = indicatorsTotalCount === 0;
+  const installedTIIntegrations: Integration[] =
```

Comment:
let's add a link to the issue where the logic is described. It might not be that obvious why we check to ti_utils for example